src/features/deployEcosystem/application/convertToEcosystemMainAccount.ts

```diff
@@ -24,13 +24,62 @@ export type NormalizedEcosystemMainAccount = {
   projectReceivers: ProjectReceiver[];
 };
 
-// The `EcosystemMainAccount` structure has a two-level design:
-//  - Level 1: Contains both direct project receivers and references to sub-lists.
-//  - Level 2: Contains sub-lists of additional receivers.
-// The structure is limited by `MAX_SPLITS_RECEIVERS` per level:
-// 200 (sub-lists) × 200 (receivers per sub-list) = 40,000 accounts
 const MAX_SPLITS_RECEIVERS = 200; // Hardcoded in Drips contracts.
 const MAX_NUMBER_OF_NODES = 40000; // Calculated based on `MAX_SPLITS_RECEIVERS`.
+const NORMALIZATION_TARGET = 1_000_000;
+
+/**
+ * Helper to normalize a list of weights ensuring every entry gets at least 1
+ * and the total equals NORMALIZATION_TARGET.
+ */
+function normalizeWeights(
+  weights: number[],
+  target: number = NORMALIZATION_TARGET,
+): number[] {
+  const n = weights.length;
+  if (n === 0) return [];
+  // Guard: Ensure total raw weight > 0.
+  const totalRaw = weights.reduce((sum, w) => sum + w, 0);
+  if (totalRaw === 0) {
+    throw new Error('Cannot normalize weights: total raw weight is 0.');
+  }
+  // Each entry gets at least 1.
+  const base = 1;
+  const remainingTarget = target - n;
+  const prelim = weights.map(w => Math.floor((w / totalRaw) * remainingTarget));
+  const baseAllocations = prelim.map(p => p + base);
+  let allocated = baseAllocations.reduce((sum, a) => sum + a, 0);
+  const diff = target - allocated;
+  // Compute fractional remainders for each weight.
+  const fractions = weights.map(
+    w =>
+      (w / totalRaw) * remainingTarget -
+      Math.floor((w / totalRaw) * remainingTarget),
+  );
+  // Distribute any remaining weight one by one to those with the highest fraction.
+  const indicesSorted = Array.from({length: n}, (_, i) => i).sort(
+    (i, j) => fractions[j] - fractions[i],
+  );
+  for (let i = 0; i < diff; i++) {
+    baseAllocations[indicesSorted[i]] += 1;
+  }
+  allocated = baseAllocations.reduce((sum, a) => sum + a, 0);
+  if (allocated !== target) {
+    logger.error(
+      `Normalization error: allocated total ${allocated} does not equal target ${target}.`,
+    );
+    throw new Error('Normalization failed to meet target.');
+  }
+  baseAllocations.forEach((w, i) => {
+    if (w <= 0) {
```

Comment:
Should this comparison be with 1 and not 0?

Reply:
Good catch, it's confusing (but it's correct). Each allocation starts with 1, so checking for `w<=0` is the "same as" `w < 1`, but of course confusing. Changing.